test_curses fails on non-UTF-8 locale

```
$ LC_ALL=en_US.iso885915 ./python -m test -vuall test_curses -m test.test_curses.TestCurses.test_output_character
== CPython 3.11.0a7+ (heads/re-group-name:5026649f98, Apr 24 2022, 22:45:28) [GCC 9.4.0]
== Linux-5.4.0-109-generic-x86_64-with-glibc2.31 little-endian
== cwd: /home/serhiy/py/cpython/build/test_python_197494�
== CPU count: 8
== encodings: locale=ISO-8859-15, FS=iso8859-15
0:00:00 load avg: 1.53 Run tests sequentially
0:00:00 load avg: 1.53 [1/1] test_curses
TERM=xterm-256color
test_output_character (test.test_curses.TestCurses.test_output_character) ... 

======================================================================
FAIL: test_output_character (test.test_curses.TestCurses.test_output_character)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/serhiy/py/cpython/Lib/test/test_curses.py", line 270, in test_output_character
    with self.assertRaises((UnicodeEncodeError, OverflowError)):
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: (<class 'UnicodeEncodeError'>, <class 'OverflowError'>) not raised

----------------------------------------------------------------------
Ran 1 test in 0.006s

FAILED (failures=1)
test test_curses failed
test_curses failed (1 failure)

== Tests result: FAILURE ==

1 test failed:
    test_curses

Total duration: 98 ms
Tests result: FAILURE
```

--- a/Lib/test/test_curses.py
+++ b/Lib/test/test_curses.py
@@ -268,7 +268,12 @@ class TestCurses(unittest.TestCase):
         stdscr.echochar(b'A')
         stdscr.echochar(65)
         with self.assertRaises((UnicodeEncodeError, OverflowError)):
-            stdscr.echochar('\u20ac')
+            # Unicode is not fully supported yet, but at least it does
+            # not crash.
+            # It is supposed to fail because either the character is
+            # not encodable with the current encoding, or it is encoded to
+            # a multibyte sequence.
+            stdscr.echochar('\u0114')
         stdscr.echochar('A', curses.A_BOLD)
         self.assertIs(stdscr.is_wintouched(), False)
 

